src/Libraries/RevitNodesUI/GenericClasses.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), AstFactory.BuildNullNode()) };
+            }
+
+            // Assume we are going to cast a Revit Element
+            Type genericElementType = typeof(Autodesk.Revit.DB.Element);
+
+            // Cast the selected object to a Revit Element and get its Id
+            Autodesk.Revit.DB.ElementId Id = ((Autodesk.Revit.DB.Element)Items[SelectedIndex].Item).Id;
+
+            // Select the element using the elementIds Integer Value
+            var node = AstFactory.BuildFunctionCall("Revit.Elements.ElementSelector", "ByElementId",
+                new List<AssociativeNode> { AstFactory.BuildIntNode(Id.IntegerValue) });
+
+            // Return the selected element
+            return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), node) };
+        }
+    }
+
+    /// <summary>
+    /// Generic UI Dropdown node baseclass for Enumerations.
+    /// This class populates a dropdown with all enumeration values of the specified type.
+    /// </summary>
+    public abstract class CustomGenericEnumerationDropDown : RevitDropDownBase
+    {
+        /// <summary>
+        /// Generic Enumeration Dropdown
+        /// </summary>
+        /// <param name="name">Node Name</param>
+        /// <param name="enumerationType">Type of Enumeration to Display</param>
+        public CustomGenericEnumerationDropDown(string name, Type enumerationType) : base(name) { this.EnumerationType = enumerationType; PopulateItems(); }
+
+        /// <summary>
+        /// Type of Enumeration
+        /// </summary>
+        public Type EnumerationType;
+
+        protected override CoreNodeModels.DSDropDownBase.SelectionState PopulateItemsCore(string currentSelection)
+        {
+            PopulateItems();
+            return SelectionState.Done;
+        }
+
+        /// <summary>
+        /// Populate Items in Dropdown menu
+        /// </summary>
+        public void PopulateItems()
+        {
+            if (this.EnumerationType != null)
+            {
+                // Clear the dropdown list
+                Items.Clear();
+
+                // Get all enumeration names and add them to the dropdown menu
+                foreach (string name in Enum.GetNames(EnumerationType))
+                {
+                    Items.Add(new CoreNodeModels.DynamoDropDownItem(name, Enum.Parse(EnumerationType, name)));
+                }
+
+                Items = Items.OrderBy(x => x.Name).ToObservableCollection();
+            }
+        }
+
+        /// <summary>
+        /// Assign the selected Enumeration value to the output
+        /// </summary>
+        public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
+        {
+            // If there the dropdown is still empty try to populate it again
+            if (Items.Count == 0 || Items.Count == -1)
+            {
+                PopulateItems();
```

Comment:
if the EnumerationType is null or the enumeration is empty it would lead to an empty list of elements. But I'm not sure if BuildOutputAst is triggered on user interaction only? I guessed so. So far I has been around some time in the Rebar package without causing any infinite loops. I will add a check for null and empty enums:
```if (this.EnumerationType != null && Enum.GetNames(this.EnumerationType).Length > 0)```